### PR TITLE
Tobin more cv cleanup

### PR DESCRIPTION
**Pull request by tobine**
First 2 CLs are simple.
Third is a bit more complex, pulling data from 4 separate VkDeviceMemory maps into a single map and consolidating around the DEVICE_MEM_INFO struct.

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -546,20 +541,21 @@ static void add_mem_obj_info(layer_data *my_data, void *object, const VkDeviceMe
     my_data->memObjMap[mem].valid = false;
 }
 
-static VkBool32 validate_memory_is_valid(layer_data *my_data, VkDeviceMemory mem, const char *functionName,
+static VkBool32 validate_memory_is_valid(layer_data *dev_data, VkDeviceMemory mem, const char *functionName,
                                          VkImage image = VK_NULL_HANDLE) {
     if (mem == MEMTRACKER_SWAP_CHAIN_IMAGE_KEY) {
-        MT_OBJ_BINDING_INFO *pBindInfo = get_object_binding_info(my_data, (uint64_t)(image), VK_DEBUG_REPORT_OBJECT_TYPE_IMAGE_EXT);
+        MT_OBJ_BINDING_INFO *pBindInfo =
+            get_object_binding_info(dev_data, (uint64_t)(image), VK_DEBUG_REPORT_OBJECT_TYPE_IMAGE_EXT);
```
Comment (lentinem) [inline layers/core_validation.cpp]: reinterpret_cast

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -568,14 +564,15 @@ static VkBool32 validate_memory_is_valid(layer_data *my_data, VkDeviceMemory mem
     return false;
 }
 
-static void set_memory_valid(layer_data *my_data, VkDeviceMemory mem, bool valid, VkImage image = VK_NULL_HANDLE) {
+static void set_memory_valid(layer_data *dev_data, VkDeviceMemory mem, bool valid, VkImage image = VK_NULL_HANDLE) {
     if (mem == MEMTRACKER_SWAP_CHAIN_IMAGE_KEY) {
-        MT_OBJ_BINDING_INFO *pBindInfo = get_object_binding_info(my_data, (uint64_t)(image), VK_DEBUG_REPORT_OBJECT_TYPE_IMAGE_EXT);
+        MT_OBJ_BINDING_INFO *pBindInfo =
+            get_object_binding_info(dev_data, (uint64_t)(image), VK_DEBUG_REPORT_OBJECT_TYPE_IMAGE_EXT);
```
Comment (lentinem) [inline layers/core_validation.cpp]: same

File: layers/core_validation.cpp [hunk 3]
```diff
@@ -5800,73 +5798,71 @@ VK_LAYER_EXPORT VKAPI_ATTR void VKAPI_CALL vkDestroyImage(VkDevice device, VkIma
...
     }
 }
 
-VkBool32 validate_memory_range(layer_data *my_data, const unordered_map<VkDeviceMemory, vector<MEMORY_RANGE>> &memory,
-                               const MEMORY_RANGE &new_range, VkDebugReportObjectTypeEXT object_type) {
+VkBool32 validate_memory_range(layer_data *dev_data, const vector<MEMORY_RANGE> &ranges, const MEMORY_RANGE &new_range,
+                               VkDebugReportObjectTypeEXT object_type) {
     VkBool32 skip_call = false;
-    if (!memory.count(new_range.memory))
+    if (ranges.empty())
```
Comment (lentinem) [inline layers/core_validation.cpp]: This is not needed. The prior check was to make sure that the device memory existed in the hash.

**Comment by tobine** [comment]
pushed to master

**Pull request closed without merging**